### PR TITLE
One fix per day #5

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java [hunk 1]
```diff
@@ -236,7 +236,6 @@ private void internalStore(Mail mc) throws MessagingException, IOException {
             keys.add(key);
         }
```
Comment (remk) [inline server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java]: typo in commit name s/varialbe/variable

File: server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java [hunk 2]
```diff
@@ -265,13 +264,12 @@ private void internalStore(Mail mc) throws MessagingException, IOException {
                 }
 
                 // its an update
```
Comment (remk) [inline server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java]: this comment could be removed too

**Comment by chibenwa** [comment]
`HeadersToHTTPTest.shouldBeFailedWhenServiceNotExists:98 » IllegalArgument port...`

unrelated.

test this please

**Comment by remk** [comment]
be wary to rename the fixup `fixup! [Refactoring] update varialbe is always true` to `fixup! [Refactoring] update variable is always true` if you want it to be applied to the good commit during autosquash.

**Comment by chibenwa** [comment]
> be wary to rename the fixup fixup! [Refactoring] update varialbe is always true to fixup! [Refactoring] update variable is always true if you want it to be applied to the good commit during autosquash.

Don't worry, I never automatically auto-squash

File: server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java [hunk 3]
```diff
@@ -245,7 +299,42 @@ public Mail retrieve(MailKey key) throws MessagingException {
...
 
     @Override
-    protected void internalRemove(MailKey key) throws MessagingException {
+    public void remove(Mail mail) throws MessagingException {
+        remove(MailKey.forMail(mail));
+    }
+
+    @Override
+    public void remove(Collection<Mail> mails) throws MessagingException {
+        for (Mail mail : mails) {
```
Comment (blackheaven) [inline server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java]: mails.forEach?

File: server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java [hunk 4]
```diff
@@ -245,7 +299,42 @@ public Mail retrieve(MailKey key) throws MessagingException {
...
+    }
+
+    @Override
+    public long size() {
+        return Iterators.size(list());
+    }
+
+    @Override
+    public void removeAll() {
+        ImmutableList.copyOf(list())
```
Comment (blackheaven) [inline server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java]: Why a copy here?

File: server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java [hunk 5]
```diff
@@ -263,10 +352,7 @@ protected void internalRemove(MailKey key) throws MessagingException {
                 clone = new ArrayList<>(keys);
             }
         } else {
-            clone = new ArrayList<>();
-            for (Iterator<String> i = objectRepository.list(); i.hasNext(); ) {
-                clone.add(i.next());
-            }
+            clone = objectRepository.list().collect(Collectors.toCollection(ArrayList::new));
```
Comment (blackheaven) [inline server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java]: `Guavate.toImmutableList`

File: server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java (hunk 3 above)
Comment (chibenwa) [reply server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java]: @blackheaven  can you contribute these fixes?

File: server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java [hunk 6]
```diff
@@ -344,6 +342,11 @@ private void internalRemove(MailKey key) {
 
     @Override
     public Iterator<MailKey> list() {
+        return listStream()
+            .iterator();
+    }
+
+    public Stream<MailKey> listStream() {
```
Comment (mbaechler) [inline server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java]: public?

**Comment by mbaechler** [comment]
```

[37b7559edeeee1603d13f31446923273e2e63c4d] [ERROR] Errors: 
[37b7559edeeee1603d13f31446923273e2e63c4d] [ERROR]   ConcurrentTestRunnerTest.runShouldPerformAllOperationsEvenOnOccasionalExceptions:230 » NotTerminated
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**